count towards precreated images

--- a/upsies/jobs/screenshots.py
+++ b/upsies/jobs/screenshots.py
@@ -49,9 +49,9 @@ class ScreenshotsJob(base.JobBase):
 
         :param precreated: Sequence of paths of already existing screenshots
 
-            These do not count towards the wanted number of screenshots.
-            `count` screenshots are created in addition to any precreated
-            screenshots.
+            If `count` is given, no more than `count` screenshots are created
+            in total. If there are more than `count` precreated screenshots,
+            only the first `count` are used.
 
         :param exclude_files: Sequence of glob patterns (:class:`str`) and
             :class:`re.Pattern` objects (return value from :func:`re.compile`)
@@ -380,6 +380,11 @@ def _screenshots_process(
         precreated, exclude_files, timestamps, count,
         from_all_videos, output_dir, overwrite, tonemap,
 ):
+    # Determine how many precreated screenshots to use
+    if count:
+        precreated = precreated[:count]
+        count = max(0, count - len(precreated))
+
     # First we have to collect all video files we want to create screenshots from. This allows us to
     # report the total number of screenshots.
     screenshot_infos = _get_screenshot_infos(
@@ -521,7 +526,7 @@ def _validate_timestamps(*, video_filepath, timestamps, count):
     # Deduplicated validated_timestamps.
     validated_timestamps = sorted(set(validated_timestamps))
 
-    if not timestamps and not count:
+    if not timestamps and count is None:
         count = DEFAULT_NUMBER_OF_SCREENSHOTS
 
     # Add more timestamps if the user didn't specify less than `count`.
