hardcode add "upload" tags to every upsies submission

--- a/upsies/utils/btclient.py
+++ b/upsies/utils/btclient.py
@@ -147,6 +147,11 @@ class BtClient:
                     hashes=infohash,
                     category=str(self._category),
                 )
+                await self._api.call(
+                    'torrents/addTags',
+                    hashes=infohash,
+                    tags='upload',
+                )
             except aiobtclientrpc.RPCError as e:
                 if 'incorrect category name' in str(e).lower():
                     raise errors.TorrentAddError(f'Unknown category: {self._category}') from e
