ffmpeg filter to hdr tonemap screenshots

--- a/upsies/utils/image.py
+++ b/upsies/utils/image.py
@@ -70,6 +70,13 @@ def _make_screenshot_cmd(video_file, timestamp, screenshot_file):
         for filtername, filtervalue in vf.items()
     )

+    if any(
+            hdr_formats.startswith('HDR')
+            for hdr_formats in utils.mediainfo.video.get_hdr_formats(video_file)
+    ):
+        # https://ffmpeg.org/ffmpeg-filters.html#tonemap-1
+        vf += ',zscale=t=linear,tonemap=hable,zscale=t=bt709,format=rgb24'
+
     return (
         _ffmpeg_executable(),
         '-y',
